tedana/metrics/dependence.py

```diff
@@ -0,0 +1,527 @@
+    Parameters
+    ----------
+    maps : (S x C)
+    mask
+    ref_img
+    threshold
+    csize
+
+    Returns
+    -------
+    maps_thresh
+    """
+    n_voxels, n_components = maps.shape
+    maps_thresh = np.zeros([n_voxels, n_components], bool)
+    LGR.info('Performing spatial clustering of components')
+    if csize is None:
+        csize = np.max([int(n_voxels * 0.0005) + 5, 20])
+    else:
+        csize = int(csize)
+    LGR.debug('Using minimum cluster size: {}'.format(csize))
+
+    for i_comp in range(n_components):
+        # Cluster-extent threshold and binarize F-maps
+        ccimg = io.new_nii_like(
+            ref_img,
+            np.squeeze(utils.unmask(maps[:, i_comp], mask)))
+
+        maps_thresh[:, i_comp] = utils.threshold_map(
+            ccimg, min_cluster_size=csize, threshold=threshold, mask=mask,
+            binarize=True)
+    return maps_thresh
+
+
+def threshold_to_match(maps, n_sig_voxels, mask, ref_img, csize=None):
+    """
+    Cluster-extent threshold a map to have roughly some requested number of
+    significant voxels (with clusters accounted for).
+
+    Parameters
+    ----------
+    maps : (S x C)
+    n_sig_voxels
+    mask
+    ref_img
+    csize
+
+    Returns
+    -------
+    clmaps
+    """
+    n_voxels, n_components = maps.shape
+    if csize is None:
+        csize = np.max([int(n_voxels * 0.0005) + 5, 20])
+    else:
+        csize = int(csize)
+    LGR.debug('Using minimum cluster size: {}'.format(csize))
+
+    clmaps = np.zeros([n_voxels, n_components], bool)
+    for i_comp in range(n_components):
+        # Initial cluster-defining threshold is defined based on the number
+        # of significant voxels from the F-statistic maps. This threshold
+        # will be relaxed until the number of significant voxels from both
+        # maps is roughly equal.
+        ccimg = io.new_nii_like(
+            ref_img,
+            utils.unmask(stats.rankdata(maps[:, i_comp]), mask))
+        step = int(n_sig_voxels[i_comp] / 10)
+        rank_thresh = n_voxels - (n_sig_voxels[i_comp] - 1)
+        while True:
+            clmap = utils.threshold_map(
+                ccimg, min_cluster_size=csize,
+                threshold=rank_thresh, mask=mask,
+                binarize=True)
+            diff = n_sig_voxels[i_comp] - clmap.sum()
+            if diff < 0 or clmap.sum() == 0:
+                rank_thresh += step
+                clmap = utils.threshold_map(
+                    ccimg, min_cluster_size=csize,
+                    threshold=rank_thresh, mask=mask,
+                    binarize=True)
+                break
+            else:
+                rank_thresh -= step
+        clmaps[:, i_comp] = clmap
+    return clmaps
+
+
+def calculate_dependence_metrics(F_T2_maps, F_S0_maps, Z_maps):
+    """
+    Calculate Kappa and Rho metrics from F-statistic maps.
+    Just a weighted average over voxels.
+
+    Parameters
+    ----------
+    F_T2_maps, F_S0_maps : (S x C) array_like
```

Comment:
This does not appear correctly in the readthedocs API. It's: "F_S0_maps (F_T2_maps,)"